### DISCUSSION TITLE
Force Sentinel-2 1c data downloading problem using force-level1-csd function

### DISCUSSION THREAD

**Discussion by JunxiangPengSLU** [davidfrantz/force#239] [Q&A]
Hi,

Currently I have a problem for Sentinel-2 1c data downloading using force-level1-csd function . Like if you run the below script (if you run it, you should specify metadata directory, download directory and queue file), 

“ force-level1-csd -c 0,75 -d 20200301,20200630 -s S2A metadata-dir level-1-datapool queue T33VWL”.

Then it gives an error said “bin/force-level1-csd: line 84: 50,09: value too great for base (error token is "09") “ 
Same problem also happed for some other cases with “value too great for base (error token is "08") “.

I tried to google it and it looks like it is because of the [octal problem](https://stackoverflow.com/questions/12821715/convert-string-into-integer-in-bash-script-leading-zero-number-error/12821845#12821845), which does not recognize “08” and “09”, but I am not sure this is  the reason.

Do you have this problem before? How to avoid it?

Thanks,
Best ergards,
Junxiang

**Comment by ernstste** [comment]
Hi Junxiang,

I'm not able to reproduce the error on my end.
How are you running force (system, docker, type of shell)? 
Maybe you can try running the same command prefixed with `bash -x` and post the output in a code block here.

Stefan

**Comment by JunxiangPengSLU** [comment]
[downloading_issue.txt](https://github.com/davidfrantz/force/files/9890582/downloading_issue.txt)
Hi Stefan,

Thanks a lot for replying.
I run Force in the Terminal (Ctrl+Alt+T) in Ubuntu 22.04. I am very new for Linux (start to learn it from this week), so I do not know so much about this.
I have already run the code prefixed with bash -x, and the attached file is the output. If you can take a look at it and figure it out what the problem is, it will be great and highly appreciated.
Thanks a lot,

Best, 
Junxiang

**Comment by davidfrantz** [comment]
Hi both,

seems like a non-English locale issue, right? 

@JunxiangPengSLU, the most robust way is to use FORCE through a Docker (or any other container tool) because this standardizes for things like this.

Alternatively, you could try to set your system to an english locale, although I would still recommd to run via Docker.

Cheers,
David

**Comment by ernstste** [comment]
Thanks for the log.
It's exactly as @davidfrantz has pointed out, the problem is due to the decimal separator. 
I will add a fix for locales using decimal commas instead of decimal points, but as David says it would be highly recommended to run FORCE in a containerized environment. 

Stefan

**Comment by JunxiangPengSLU** [comment]
Hi David and Stefan,
Thanks a lot for the suggestion and help. Let me try to run force using Docker.

Best,
Junxiang

**Comment by ernstste** [comment]
Here is a suggestion on how you can run the docker (you can make your life easier if you add this as alias to your .bashrc file in the home directory). You will have to adjust the path after `-v` to where you want to save the Sentinel-2 data.

```
docker run \
-t \
-e FORCE_CREDENTIALS=/app/credentials \
-e BOTO_CONFIG=/app/credentials/.boto \
-v $HOME:/app/credentials \
-v /path/to/your/data:/path/to/your/data\
-w $PWD \
-u $(id -u):$(id -g) \
-t \
--rm \
davidfrantz/force:latest'
```

**Reply by JunxiangPengSLU** [reply]
thanks Stefan!

**Comment by ernstste** [comment]
For future reference: #240
